moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -174,13 +174,20 @@ class TimeOptimalTrajectoryGeneration : public TimeParameterization
                          const std::unordered_map<std::string, double>& velocity_limits,
                          const std::unordered_map<std::string, double>& acceleration_limits) const override;
 
+  /**
+   * \brief Compute a trajectory with the given number of waypoints (plus or minus 1, due to rounding)
```

Comment:
Would it make sense here to motivate why a finer-grained / coarser trajectory might be necessary and potential damage that can be caused e.g if you down-sample a long trajectory with 120 waypoints to 2? Is it correct to assume that the only guarantee that can be made here is that the start and goal state will be the same? The path characteristics between those two points can change completely

Reply:
Definitely needs a better comment.

Reply:
This part of your comment is actually an incorrect assumption:

> Is it correct to assume that the only guarantee that can be made here is that the start and goal state will be the same? The path characteristics between those two points can change completely

Resampling doesn't change the path since it occurs after time-parameterization. You can see it here:

https://github.com/ros-planning/moveit2/blob/50b03a1674537613e2b6f26a220e78d8572f0216/moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp#L1130

Possibly @henningkayser can confirm this. I'll clarify with a better comment.

Reply:
Maybe I am missing something here, but how can that be? If you have an 'S'-shaped path in 2D with 10 waypoints and you down-sample it to 3 waypoints how can this path of 3 waypoints still describe an 'S'?